tokio/src/runtime/time/mod.rs

```diff
@@ -25,9 +26,35 @@ use crate::time::error::Error;
 use crate::time::{Clock, Duration};
 use crate::util::WakeList;
 
+use crate::loom::sync::atomic::AtomicU64;
 use std::fmt;
 use std::{num::NonZeroU64, ptr::NonNull};
 
+struct AtomicOptionNonZeroU64(AtomicU64);
+
+// A helper type to store the `next_wake`.
+impl AtomicOptionNonZeroU64 {
+    fn new(val: Option<u64>) -> Self {
+        Self(AtomicU64::new(Self::turn(val)))
+    }
+
+    fn store(&self, val: Option<u64>) {
+        self.0.store(Self::turn(val), Ordering::Relaxed);
```

Comment:
This stores `Some(1)` if passed `Some(0)`. Can you add a comment that explains why?

Actually, it might be more clear to change this to `Option<NonZeroU64>` and move the call to `turn` to where you call this method.

Reply:
I add the helper function `next_wake_time` instead and some explains about it.